### PR TITLE
Bug 1893351: Determine Secret type upon the save

### PR DESCRIPTION
**Pull request by jhadvig**
We have been passing "Opaque" type when updating secret via Generic Secret form, even though  `kubernets/tls` and `kubernets/service-account-token` Secret types are being updated with this form type.
Since with Generic form any Secret type can be created, determine the type when the Secret is being created.

/assign @spadgett

### CODE REVIEW

File: frontend/public/components/secrets/create-secret.tsx [hunk 1]
```diff
@@ -101,6 +101,24 @@ const generateSecret = () => {
   return s4() + s4() + s4() + s4();
 };
 
+const determineSecretType = (stringData): SecretType => {
```
Comment (spadgett) [inline frontend/public/components/secrets/create-secret.tsx:104]: `stringData` makes it sounds like the parameter is a string, but it appears to be an object.

**Review by spadgett — CHANGES_REQUESTED** [review]
We shouldn't try to infer the type on edit. We should simply preserve whatever was previously in the secret. The field is immutable, so it shouldn't change. It also works for any secret type, including unknown types.

File: frontend/public/components/secrets/create-secret.tsx (hunk 1 above)
Comment (jhadvig) [reply frontend/public/components/secrets/create-secret.tsx:104]: @spadgett so the `stringData` is name of the [field](https://kubernetes.io/docs/concepts/configuration/secret/#overview-of-secrets). Similar to `data` but this take an arbitrary string value.

**Comment by jhadvig** [comment]
> We shouldn't try to infer the type on edit. We should simply preserve whatever was previously in the secret. The field is immutable, so it shouldn't change. It also works for any secret type, including unknown types.

So this fix is will take the `type` of the edited Secret. Wont edit anything.

**Review by spadgett — APPROVED** [review]
/lgtm

**Pull request merged**